src/client/App.jsx

```diff
@@ -12,41 +12,80 @@ function App() {
   useEffect(() => {
     fetch(`${apiUrl}/movie`)
       .then(res => res.json())
-      .then(res => setMovies(res.data));
+      .then(res => setMovies(res.data))
+
   }, []);
 
-  /**
-   * HINTS!
-   * 1. This handle___ functions below use async/await to handle promises, but the
-   * useEffect above is using .then to handle them. Both are valid approaches, but
-   * we should ideally use one or the other. Pick whichever you prefer.
-   *
-   * 2. The default method for the `fetch` API is to make a GET request. To make other
-   * types of requests, we must provide an object as the second argument of `fetch`.
-   * The values that you must provide are:
-   * - method
-   * - headers
-   * - body (if needed)
-   * For the "headers" property, you must state the content type of the body, i.e.:
-   *   headers: {
-   *     'Content-Type': 'application/json'
-   *   }
-   * */
 
   const handleRegister = async ({ username, password }) => {
+    const data = { username, password }
+
+    const options = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(data)
+    }
 
+    localStorage.removeItem("token")
+    const registerUserData = await fetch(`${apiUrl}/user/register`, options)
+    const registeredUser = await registerUserData.json()
+    return alert(registeredUser.message)
   };
 
+
   const handleLogin = async ({ username, password }) => {
+    const data = {
+      username,
+      password,
+    }
+
+    const options = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify(data)
+    }
+
+    const userData = await fetch(`${apiUrl}/user/login`, options)
+    const newLogin = await userData.json()
+    const userToken = newLogin.token
+
+    localStorage.setItem("token", userToken)
+    return alert(newLogin.message)
 
   };
 
+
   const handleCreateMovie = async ({ title, description, runtimeMins }) => {
+    const data = { title, description, runtimeMins }
 
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        'Authorization': `Bearer ${localStorage.getItem('token')}`
```

Comment:
It's a good idea to get the token out of local storage before using it like this. With the code as it is currently, you know know if the token is defined or not.

Reply:
Ah ok, so making a variable and setting some guard clauses around it before using it here?

Reply:
Yeah, exactly!